desktop/src/main/nl/tudelft/ti2806/riverrush/graphics/entity/Animal.java

```diff
@@ -46,6 +46,7 @@ public AnimalActor getActor() {
      */
     public void setActor(final AnimalActor aActor) {
         this.actor = aActor;
+        this.actor.setAnimal(this);
```

Comment:
Well, the problem is that collisions are checked inside the Actor itself, but the behaviour triggered by a collision is handled by the Animal. Because of this mismatch, we need some way to relate an Actor to an Animal after it has collided. The alternative is a O(n) loop over all animals to check whether they belong to the collided Actor. I'm no fan of either options, but think this is the least bad one. Do you have any other ideas?